README.md

```diff
@@ -60,15 +58,41 @@ If no `observedAttributes` are specified, they will be inferred from the keys of
 ```js
 // Other option: use PropTypes:
 function FullName(props) {
-  return <span>{props.first} {props.last}</span>
+	return (
+		<span>
+			{props.first} {props.last}
+		</span>
+	);
 }
 FullName.propTypes = {
-  first: Object,   // you can use PropTypes, or this
-  last: Object     // trick to define untyped props.
+	first: Object, // you can use PropTypes, or this
+	last: Object, // trick to define untyped props.
 };
 register(FullName, 'full-name');
 ```
 
+### Using the shadow DOM
+
+It is possible to attach a shadow root to the custom element to have your component rendered in a separate subtree.
+Customization of the shadow root `mode` is also possible on element basis.
+Read more about the shadow DOM on [MDN](https://developer.mozilla.org/en-US/docs/Web/API/ShadowRoot).
+
+```tsx
+function WithShadow(props) {
+	return <span>Hello from shadow root!</span>;
+}
+
+// use shadow root in "open" mode
+register(FullName, 'full-name', [], {
+	shadow: true,
+});
+
+// use shadow root in "closed" mode
+register(FullName, 'full-name-encapsulated', [], {
+	shadow: true,
+	mode: 'closed',
+});
+```
```

Comment:
Mind extracting this (& the formatting changes, I'm guessing they're done by the precommit hooks) to a separate PR? Helps keep things focused

Reply:
Of course, I'll do that 😄